src/test/java/uk/gov/pay/ledger/transaction/dao/TransactionDaoIT.java

```diff
@@ -505,4 +510,91 @@ void shouldReturnTransactionsCorrectlyForDateRangesAndNumberOfTransactions() {
+        void shouldRedactAllPIIFromTransactionCorrectly() throws JsonProcessingException {
+
+            JsonObject transactionDetails1 = new JsonObject();
+            transactionDetails1.addProperty("reference", "ref-1");
+            transactionDetails1.addProperty("cardholder_name", "Joe B");
+            transactionDetails1.addProperty("email", "test@email.com");
+            transactionDetails1.addProperty("address_line1", "line 1");
+            transactionDetails1.addProperty("address_line2", "line 2");
+
+            TransactionEntity transactionToRedact = aTransactionFixture()
+                    .withReference("ref-1")
+                    .withEmail("test@email.com")
+                    .withCardholderName("Joe B")
+                    .withTransactionDetails(transactionDetails1.toString())
+                    .insert(rule.getJdbi())
+                    .toEntity();
+
+            JsonObject transactionDetails2 = new JsonObject();
+            transactionDetails2.addProperty("reference", "ref-2");
+            transactionDetails2.addProperty("cardholder_name", "Jane D");
+            transactionDetails2.addProperty("email", "test2@email.com");
+            TransactionEntity transactionThatShouldNotBeRedacted = aTransactionFixture()
+                    .withReference("ref-2")
+                    .withEmail("test2@email.com")
+                    .withCardholderName("Jane D")
+                    .withTransactionDetails(transactionDetails2.toString())
+                    .insert(rule.getJdbi())
+                    .toEntity();
+
+            transactionDao.redactPIIFromTransaction(transactionToRedact.getExternalId());
+
+            TransactionEntity transactionEntityRedacted = transactionDao.findTransactionByExternalId(transactionToRedact.getExternalId()).get();
+
+            assertThat(transactionEntityRedacted.getReference(), is("<REDACTED>"));
+            assertThat(transactionEntityRedacted.getCardholderName(), is("<REDACTED>"));
+            assertThat(transactionEntityRedacted.getEmail(), is("<REDACTED>"));
+
+            JsonNode jsonNode = objectMapper.readTree(transactionEntityRedacted.getTransactionDetails());
+
+            assertThat(jsonNode.get("reference"), is(Matchers.nullValue()));
+            assertThat(jsonNode.get("cardholder_name"), is(nullValue()));
+            assertThat(jsonNode.get("email"), is(nullValue()));
+            assertThat(jsonNode.get("address_line1").asText(), is("<REDACTED>"));
+            assertThat(jsonNode.get("address_line2").asText(), is("<REDACTED>"));
+
+            TransactionEntity transactionEntityNotRedacted = transactionDao.findTransactionByExternalId(transactionThatShouldNotBeRedacted.getExternalId()).get();
+            assertThat(transactionEntityNotRedacted.getReference(), is("ref-2"));
+            assertThat(transactionEntityNotRedacted.getCardholderName(), is("Jane D"));
+            assertThat(transactionEntityNotRedacted.getEmail(), is("test2@email.com"));
```

Comment:
It would help with readability if this were another test case

Reply:
I am trying to assert that if another transaction is present it should not be redacted (that the = in SQL works for a single tx). Will add a new test for it

Reply:
Hmm given your [explanation](https://github.com/alphagov/pay-ledger/pull/2894#discussion_r1332882033), I think it's fine to leave this test as is.